src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1155,6 +1134,19 @@ public void run() {
         }
     }
 
+    private void electionAndSetCurVote() {
```

Comment:
IMHO,If you really want to refactor the code block(**Line1137-Line1149**)  **,throws Exception** may be better than catching exception in this function!

Reply:
enen, i think the exception can be appropriately handled in this function, why need to throw the exception to  caller?  :)  thanks